ugs-core/src/com/willwinder/universalgcodesender/AbstractController.java

```diff
@@ -868,7 +868,7 @@ public void addListener(ControllerListener listener) {
     @Override
     public void removeListener(ControllerListener listener) {
         if (this.listeners.contains(listener)) {
-            this.listeners.remove(listener);
+            this.listeners.removeIf(l -> l == listener);
```

Comment:
Oh, I forgot to mention why I did this. It threw ConcurrentModificationException while shutting down... sometimes. This change got rid of those errors. I'll make sure to add a comment for this.